src/test/java/picard/sam/AddOATagTest.java

```diff
@@ -0,0 +1,110 @@
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+import htsjdk.samtools.SAMRecord;
+import htsjdk.samtools.SAMRecordIterator;
+import htsjdk.samtools.SamReaderFactory;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+
+public class AddOATagTest {
+    private static final String TEST_DIR = "testdata/picard/sam/";
+    private static final String OA_TEST_DIR = TEST_DIR + "AddOATag/";
+
+    @DataProvider(name="testOAData")
+    Object[][] testOAData() {
+        return new Object[][]{
+                new Object[]{new File(OA_TEST_DIR + "aligned_withoutOA.sam"), new File(OA_TEST_DIR + "aligned_withOA.sam")},
+        };
+    }
+
+    @DataProvider(name="testIntervalListData")
+    Object[][] testIntervalListData() {
+        return new Object[][]{
+                new Object[]{new File(TEST_DIR + "aligned.sam"), null, new File(OA_TEST_DIR + "aligned_no_intervals.sam")},
+                new Object[]{new File(TEST_DIR + "aligned.sam"), new File("testdata/picard/analysis/directed/CollectHsMetrics/chrM.interval_list"), new File(TEST_DIR + "aligned.sam")},
+                new Object[]{new File(TEST_DIR + "aligned.sam"), new File(TEST_DIR + "contiguous.interval_list"), new File(OA_TEST_DIR + "aligned_with_intervals.sam")}
+        };
+    }
+
+    @Test(dataProvider = "testOAData")
+    public void testWritingOATag(final File testSam, final File truthSam) throws IOException {
+        final File clpOutput = File.createTempFile("AddOATag", ".bam");
+        clpOutput.deleteOnExit();
+
+        runAddOATag(testSam, clpOutput, null);
+
+        validateOATag(clpOutput, truthSam);
+    }
+
+    @Test(dataProvider = "testIntervalListData")
+    public void testIntervalList(final File inputSam, final File intervalList, final File truthSam) throws IOException {
+        final File clpOutput = File.createTempFile("AddOATag", ".bam");
+        clpOutput.deleteOnExit();
+
+        runAddOATag(inputSam, clpOutput, intervalList);
+
+        validateOATag(clpOutput, truthSam);
+    }
+
+    private void runAddOATag(final File inputSam, final File output, final File intervalList) throws IOException {
+        final ArrayList<String> args = new ArrayList<String>(){
+            {
+                add("INPUT=" + inputSam);
+                add("OUTPUT=" + output);
+            }
+        };
+        if (intervalList != null) {
+            args.add("INTERVAL_LIST=" + intervalList);
+        }
+        AddOATag addOATag = new AddOATag();
+        Assert.assertEquals(addOATag.instanceMain(args.toArray(new String[args.size()])), 0, "Running addOATag did not succeed");
```

Comment:
In general it's better to pass an empty array in instead of a preallocated array
```suggestion
        Assert.assertEquals(addOATag.instanceMain(args.toArray(new String[0])), 0, "Running addOATag did not succeed");
```